requirements.txt

```diff
@@ -0,0 +1,13 @@
+--index-url https://pypi.python.org/simple/
+
+# Unit test stuff
+pytest==3.2.2
```

Comment:
>=?

Reply:
What's the question?
It must be `==` not `=`, if that's what you mean.

The way this works is really primitive. IIUC the file is just cleared of comments, then the lines are joined with spaces instead of newlines, and then appended to `pip install`. That's it.

Reply:
I'm more leaning towards >= something, that was my point

Reply:
That's something for `requires` in `setup.py`. The `requirements.txt` should specify pinned versions of dependencies that make dependency installation predictable. I see the file as useful mostly in two scenarios:

- For CI, to avoid breakage when a dependency introduces a breaking change
- For bug reports, so we can ask "If you install dependencies using `pip install -r requirements.txt` does it still not work?"

Some people generate the file using `pip freeze`, which will also result in pinned versions. [Here](https://pip.readthedocs.io/en/1.1/requirements.html) is more to read on the topic.